src/main/resources/assets/js/components/AppModalComponent.jsx

```diff
@@ -31,17 +32,47 @@ var AppModalComponent = React.createClass({
     rollBackApp: React.PropTypes.func.isRequired,
     scaleApp: React.PropTypes.func.isRequired,
     suspendApp: React.PropTypes.func.isRequired,
-    tasksFetchState: React.PropTypes.number.isRequired
+    tasksFetchState: React.PropTypes.number.isRequired,
+    router: React.PropTypes.object.isRequired
   },
 
   getInitialState: function () {
+    var appid = this.props.model.get("id");
+    var activeTabId;
+
+    this.tabs = _.reduce(tabsTemplate, function (current, tab) {
+      var id = tab.id.replace(":appid", appid);
+      if (activeTabId == null) {
+        activeTabId = id;
+      }
+      current.push({
+        id: id,
+        text: tab.text
+      });
+      return current;
+    }, []);
+
     return {
       activeViewIndex: 0,
-      activeTabId: tabs[0].id,
+      activeTabId: activeTabId,
       selectedTasks: {}
     };
   },
 
+  componentDidMount: function () {
+    this.props.router.on("route:apps", function (appid, view) {
```

Comment:
I mean could this be done in Marathon.jsx's `componentDidMount` in stead and the necessary data pushed down to this component?